server/service/src/login.rs

```diff
@@ -133,6 +135,24 @@ impl LoginService {
             }
         };
 
+        let log_service = LogRowRepository::new(&service_ctx.connection);
+        let log = LogRow {
+            id: uuid(),
+            log_type: LogType::UserLoggedIn,
+            user_id: Some(user_account.id.clone()),
+            record_id: None,
+            created_datetime: Utc::now().naive_utc(),
+        };
+        match log_service.upsert_one(&log) {
+            Ok(_) => (),
+            Err(e) => {
+                return Err(LoginError::InternalError(format!(
+                    "Couldn't log login for {}: {:?}",
+                    user_account.username, e
+                )));
+            }
```

Comment:
There is an `impl From<RepositoryError> for LoginError` so you can just do `log_service.upsert_one(&log)?;`

I would do `LogRowRepository::new(&service_ctx.connection).upsert_one(&LogRow { ... } )?;` to keep the code logging compact.